sdks/go/workflow.go

```diff
@@ -0,0 +1,608 @@
+	concurrency            []*types.Concurrency
+	rateLimits             []*types.RateLimit
+	isDurable              bool
+	parents                []create.NamedTask
+	waitFor                condition.Condition
+	skipIf                 condition.Condition
+}
+
+// WithRetries sets the number of retry attempts for failed tasks.
+func WithRetries(retries int) TaskOption {
+	return func(config *taskConfig) {
+		config.retries = int32(retries)
+	}
+}
+
+// WithRetryBackoff configures exponential backoff for task retries.
+func WithRetryBackoff(factor float32, maxBackoffSeconds int) TaskOption {
+	return func(config *taskConfig) {
+		config.retryBackoffFactor = factor
+		config.retryMaxBackoffSeconds = int32(maxBackoffSeconds)
+	}
+}
+
+// WithTimeout sets the maximum execution duration for a task.
+func WithTimeout(timeout time.Duration) TaskOption {
+	return func(config *taskConfig) {
+		config.executionTimeout = timeout
+	}
+}
+
+// WithCron configures standalone tasks to run on a cron schedule.
+// Only applicable to standalone tasks, not workflow tasks.
+func WithCron(cronExpressions ...string) TaskOption {
+	return func(config *taskConfig) {
+		config.onCron = cronExpressions
+	}
+}
+
+// WithEvents configures standalone tasks to trigger on specific events.
+// Only applicable to standalone tasks, not workflow tasks.
+func WithEvents(events ...string) TaskOption {
+	return func(config *taskConfig) {
+		config.onEvents = events
+	}
+}
+
+// WithFilters sets default filters for event-triggered tasks.
+func WithFilters(filters ...types.DefaultFilter) TaskOption {
+	return func(config *taskConfig) {
+		config.defaultFilters = filters
+	}
+}
+
+// WithConcurrency sets concurrency limits for task execution.
+func WithConcurrency(concurrency ...*types.Concurrency) TaskOption {
+	return func(config *taskConfig) {
+		config.concurrency = concurrency
+	}
+}
+
+// WithDurable marks a task as durable, enabling persistent state and long-running operations.
+func WithDurable() TaskOption {
+	return func(config *taskConfig) {
+		config.isDurable = true
+	}
+}
+
+// WithRateLimits sets rate limiting for task execution.
+func WithRateLimits(rateLimits ...*types.RateLimit) TaskOption {
+	return func(config *taskConfig) {
+		config.rateLimits = rateLimits
+	}
+}
+
+// WithParents sets parent task dependencies.
+func WithParents(parents ...create.NamedTask) TaskOption {
+	return func(config *taskConfig) {
+		config.parents = parents
+	}
+}
+
+// WithWaitFor sets a condition that must be met before the task executes.
+func WithWaitFor(condition condition.Condition) TaskOption {
+	return func(config *taskConfig) {
+		config.waitFor = condition
+	}
+}
+
+// WithSkipIf sets a condition that will skip the task if met.
+func WithSkipIf(condition condition.Condition) TaskOption {
+	return func(config *taskConfig) {
+		config.skipIf = condition
+	}
+}
+
+// Task represents a task reference for building DAGs.
+type Task struct {
+	NamedTask create.NamedTask
```

Comment:
Do we just want to give it a `Name()` method instead? I think we needed this pattern for generics but I don't see why we need it atm